### PR TITLE
[CARBONDATA-1152] List partition not working properly

### PR DESCRIPTION
**Pull request by BJangir**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[CARBONDATA-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).
 - [ ] Testing done
 
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - What manual testing you have done?
        - Any additional information to help reviewers in testing this change.
         
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
                 
---

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/208/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2331/

**Comment by BJangir** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2374/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/253/

**Comment by ravipesala** [comment]
@BJangir Can you add testcase for it.

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala [hunk 1]
```diff
@@ -173,9 +173,20 @@ class CarbonSqlAstBuilder(conf: SQLConf) extends SparkSqlAstBuilder(conf) {
         s"Values must be specified for key(s): ${ badKeys.mkString("[", ",", "]") }", ctx)
     }
     props.map { case (key, value) =>
-      (key.toLowerCase, value.toLowerCase)
+      if (needToConvertToLowerCase(key)) {
+          (key.toLowerCase, value.toLowerCase)
+        }
+        else
```
Comment (QiangCai) [inline integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala]: code style

**Comment by QiangCai** [comment]
@BJangir Spark 1.6 has same issue. can you check CarbonDDLSqlParser and fix it?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2451/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/332/

File: integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala [hunk 2]
```diff
@@ -774,11 +774,17 @@ abstract class CarbonDDLSqlParser extends AbstractCarbonSparkSQLParser {
    * @return
    */
   protected def getProperties(node: Node): Seq[(String, String)] = {
+    val noConvertList = Array("LIST_INFO", "RANGE_INFO")
```
Comment (QiangCai) [inline integration/spark-common/src/main/scala/org/apache/spark/sql/catalyst/CarbonDDLSqlParser.scala]: remove unused variable

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2464/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/348/<h2>Failed Tests: <span class='status-failure'>1</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-core' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/348/org.apache.carbondata$carbondata-core/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-core</a>: <span class='status-failure'>1</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/348/org.apache.carbondata$carbondata-core/testReport/org.apache.carbondata.core.scan.filter.executer/ExcludeFilterExecuterImplTest/testPerformance/'><strong>org.apache.carbondata.core.scan.filter.executer.ExcludeFilterExecuterImplTest.testPerformance</strong></a></li></ul>

**Comment by BJangir** [comment]
retest this please

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/352/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2468/

**Comment by QiangCai** [comment]
LGTM

**Pull request closed without merging**